transit_odp/publish/tests/test_seasonal_service_views.py

```diff
@@ -11,6 +14,133 @@
 pytestmark = pytest.mark.django_db
 
 
+class TestSeasonalServiceView:
+    NUMBER_SEASONAL_SERVICE_PER_ORGANISATION = 15
+
+    def setup(self):
+        self.user = UserFactory.create(account_type=AccountType.org_staff.value)
+        self.org = self.user.organisation
+        self.licence = BODSLicenceFactory(organisation=self.org)
+        for i in range(self.NUMBER_SEASONAL_SERVICE_PER_ORGANISATION):
+            SeasonalServiceFactory(licence=self.licence)
+        self.url = reverse(
+            "seasonal-service", host=PUBLISH_HOST, kwargs={"pk1": self.org.id}
+        )
+
+    def test_create_table_seasonal_service(self, publish_client):
+        """
+        GIVEN : a NUMBER_SEASONAL_SERVICE_PER_ORGANISATION number of seasonal services
+        WHEN  : we open the `seasonal-service' link
+        THEN  : the data should be appear in a table, with the columns:
+               `Licence number'	`Service code' `Service begins' `Service ends' `Actions'
+               and the number of rows has to be equal to
+               NUMBER_SEASONAL_SERVICE_PER_ORGANISATION
+        """
+        publish_client.force_login(user=self.user)
+        response = publish_client.get(self.url)
+
+        assert response.status_code == 200
+        assert response.context.get("table").paginator.num_pages == 2
+        assert (
+            response.context.get("table").data.model.objects.count()
+            == self.NUMBER_SEASONAL_SERVICE_PER_ORGANISATION
+        )
+
+    def test_delete_seasonal_service(self, publish_client):
+        """
+        GIVEN : a NUMBER_SEASONAL_SERVICE_PER_ORGANISATION number of seasonal services
+        WHEN  : we open the `seasonal-service' link
+        THEN  : the data should be appear in a table, with the columns:
+               `Licence number'	`Service code' `Service begins' `Service ends' `Actions'
+               and the number of rows has to be equal to
+               NUMBER_SEASONAL_SERVICE_PER_ORGANISATION
+               then we delete the first seasonal service and the number of the rows
+               has to be equal to NUMBER_SEASONAL_SERVICE_PER_ORGANISATION - 1
+        """
+        publish_client.force_login(user=self.user)
+        response = publish_client.get(self.url)
+
+        assert response.status_code == 200
+        assert response.context.get("table").paginator.num_pages == 2
+        assert (
+            response.context.get("table").data.model.objects.count()
+            == self.NUMBER_SEASONAL_SERVICE_PER_ORGANISATION
+        )
+        delete_url = reverse(
+            "delete-seasonal-service",
+            host=PUBLISH_HOST,
+            kwargs={"pk1": self.org.id},
+        )
+
+        seasonal_services = SeasonalService.objects.first()
+        delete_data = {"id": seasonal_services.id}
+        publish_client.post(
+            delete_url,
+            data=delete_data,
+        )
+        assert (
```

Comment:
Dont forget to assert which page you are on. This is REALLY important to you!!

Reply:
in this test first page, here I testing `None`, I'll add new test case!!